src/codeu/chat/server/Controller.java

```diff
@@ -56,6 +59,326 @@ public ConversationHeader newConversation(String title, Uuid owner) {
+      Integer setCount = (currentCount + count < 0) ? 0 : currentCount + count;
+
+      foundConvo.unseenMessages.put(foundUser.id, setCount);
+
+      result = foundConvo.unseenMessages.get(foundUser.id);
+
+      LOG.info(
+              "updateUsersUnseenMessagesCount success (user.id=%s conversation.id=%s count=%s)",
+              foundUser.id,
+              foundConvo.id,
+              setCount);
+    } else {
+      LOG.info(
+              "updateUsersUnseenMessagesCount fail (user.id=%s conversation.id=%s)",
+              foundUser.id,
+              foundConvo.id);
+    }
+
+    return result;
+  }
+
+  @Override
+  public Time updateUsersLastStatusUpdate(Uuid user, Time time){
+    final User foundUser = model.userById().first(user);
+
+    Time update = null;
+
+    if(foundUser != null && time.inMs() > foundUser.creation.inMs()){
+      foundUser.lastStatusUpdate = time;
+      update = foundUser.lastStatusUpdate;
+
+      LOG.info(
+              "updateUsersLastStatusUpdate success (user.id=%s time=%s)",
+              foundUser.id,
+              time.inMs());
+    } else {
+
+      LOG.info(
+              "updateUsersLastStatusUpdate fail (user.id=%s time=%s)",
+              foundUser.id,
+              time.inMs());
+    }
+
+    return update;
+  }
+
+  @Override
+  public Map<Uuid, Time> newUpdatedConversation(Uuid user, Uuid convo, Time time){
+    final User foundUser = model.userById().first(user);
+    final ConversationHeader foundConversation = model.conversationById().first(convo);
+
+    Map<Uuid, Time> map = null;
+
+    if(foundUser != null && foundConversation != null){
+      foundUser.updatedConversations.put(foundConversation.id, time);
+      map = foundUser.updatedConversations;
+
+      LOG.info(
+              "newUpdatedConversation success (user.id=%s conversation.id=%s time=%s)",
+              foundUser.id,
+              foundConversation.id,
+              time.inMs());
+    } else {
+      LOG.info(
+              "newUpdatedConversation fail (user.id=%s conversation.id=%s time=%s)",
+              foundUser.id,
+              foundConversation.id,
+              time.inMs());
+    }
+
+    return map;
+  }
+
+  @Override
+  public Collection<Uuid> newUserInterest(Uuid user1, Uuid user2){
+    final User foundUser = model.userById().first(user1);
+    final User followedUser = model.userById().first(user2);
+
+    if(foundUser != null && followedUser != null) {
+      foundUser.userInterests.add(followedUser.id);
+
+      LOG.info(
+              "newUserInterest success (user.id=%s user.id=%s)",
+              foundUser.id,
+              followedUser.id);
+    } else {
+      LOG.info(
+              "newUserInterest fail - user/followed user doesn't exist (user.id=%s user.id=%s)",
+              foundUser.id,
+              followedUser.id);
+    }
+
+    return foundUser.userInterests;
```

Comment:
Yup, noticed this while doing the access controls and forgot to fix it. Thanks! All nulls have been checked for this server's Controller class :)